overlord/ifacestate/helpers.go

```diff
@@ -211,10 +211,13 @@ func (m *InterfaceManager) reloadConnections(snapName string) ([]string, error)
 			continue
 		}
 		if err := m.repo.Connect(connRef); err != nil {
+			// FIXME: we may encounter an error here if there is a stray connection in the state,
+			// they should be cleaned up
 			logger.Noticef("%s", err)
```

Comment:
Can you restructure the code a little so that Connect can say "no such thing connect" and then we:
1) don't log the error
2) don't mark the snap as affected so no security setup is done

Reply:
Done.